src/library/VaultLib.sol

```diff
@@ -199,7 +206,17 @@ library VaultLib {
     {
         uint256 totalAssets = IVault(address(this)).totalAssets();
         uint256 totalSupply = getERC20Storage().totalSupply;
-        baseAssets = shares.mulDiv(totalAssets + 1, totalSupply + 1, rounding);
+
+        // Handle the case when totalAssets or totalSupply is zero
+        if (totalAssets == 0 || totalSupply == 0) {
```

Comment:
this is still potentially problematic.

What if these conditions are hit after the initial deposit?

It implies we mint a 1:1, is that still accurate?